scripts/core/editor3/reducers/editor3.jsx

```diff
@@ -267,6 +270,10 @@ function moveBlock(state, {block, dest, insertionMode}) {
         insertionMode
     );
 
+    if (returnState) {
```

Comment:
It's much simpler when a return type is always the same. If you need a different one, create another function and put common functionality in a `base` function.

![return-type-refactoring](https://user-images.githubusercontent.com/2076953/44079395-95a2eaa6-9fb1-11e8-9f47-6cb95d851581.png)

Reply:
Agree